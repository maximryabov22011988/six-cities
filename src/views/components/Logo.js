import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import classnames from 'classnames';

const propTypes = {
  position: PropTypes.string,
  src: PropTypes.string,
  width: PropTypes.string,
  height: PropTypes.string,
  label: PropTypes.string,
  isActive: PropTypes.bool,
};

const defaultProps = {
  src: 'img/logo.svg',
  width: '81',
  height: '41',
  label: '6 cities logo',
};

class Logo extends React.PureComponent {
  render() {
    const { position, src, width, height, label, isActive } = this.props;

    const linkClasses = classnames(
      position && `${position}__logo-link`,
      isActive && `${position}__logo-link--active`,
    );
    const logoClasses = classnames(position && `${position}__logo`);

    return isActive ? (
      <div className={linkClasses}>
        <img
          className={logoClasses}
          src={src}
          width={width}
          height={height}
          alt={label}
        />
      </div>
    ) : (
      <Link to="/" className={linkClasses}>
        <img
          className={logoClasses}
          src={src}
          width={width}
          height={height}
          alt={label}
        />
      </Link>
    );
  }
}

Logo.propTypes = propTypes;
Logo.defaultProps = defaultProps;

export default Logo;
