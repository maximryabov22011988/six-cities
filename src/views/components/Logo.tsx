import * as React from 'react';
import { Link } from 'react-router-dom';
import cn from 'classnames';

interface Props {
  height?: string;
  isActive?: boolean;
  label?: string;
  position: string;
  src?: string;
  width?: string;
}

class Logo extends React.PureComponent<Props> {
  render() {
    const {
      position,
      src = 'img/logo.svg',
      width = '81',
      height = '41',
      label = '6 cities logo',
      isActive,
    } = this.props;

    const linkClasses = cn(position && `${position}__logo-link`, isActive && `${position}__logo-link--active`);
    const logoClasses = cn(position && `${position}__logo`);

    return isActive ? (
      <div className={linkClasses}>
        <img alt={label} className={logoClasses} height={height} src={src} width={width} />
      </div>
    ) : (
      <Link className={linkClasses} to="/">
        <img alt={label} className={logoClasses} height={height} src={src} width={width} />
      </Link>
    );
  }
}

export default Logo;
