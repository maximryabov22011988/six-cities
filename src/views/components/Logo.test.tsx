import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";

import Logo from './Logo';

it('Logo correctly render', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Logo position="header" />
    </BrowserRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
