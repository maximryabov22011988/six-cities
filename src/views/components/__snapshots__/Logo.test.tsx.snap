// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo correctly render 1`] = `
<a
  className="header__logo-link"
  href="/"
  onClick={[Function]}
>
  <img
    alt="6 cities logo"
    className="header__logo"
    height="41"
    src="img/logo.svg"
    width="81"
  />
</a>
`;
